autospec/pkg_integrity.py

```diff
@@ -114,7 +114,22 @@ def __init__(self, pubkey=None, home=None):
 
     def verify(self, _, tarfile, signature):
         """Validate tarfile with signature."""
-        args = self.args + ['--verify', signature, tarfile]
+        # Since autospec can only verify one signature for now, extract the
+        # first signature from the detached signature file.
+        sig_name = signature
+        packets = parse_gpg_packets(signature)
+        if len(packets) > 1:
+            # sig file may be ascii-armored, so dearmor it first...
+            args = self.args + ['--dearmor', '--output', '-', signature]
+            output, err, code = self.exec_cmd(args)
+            if code != 0:
+                return None
+            num_bytes = packets[0]["length"]
+            first_sig = output[:num_bytes]
+            with tempfile.NamedTemporaryFile(prefix="newsig-", dir=self._home, delete=False) as new_sig_file:
+                new_sig_file.write(first_sig)
+                sig_name = new_sig_file.name
+        args = self.args + ['--verify', sig_name, tarfile]
```

Comment:
Do we want to do anything with the new_sig_file since it is generated like check it into the repo?

Reply:
I'm not sure, hmm.

Reply:
Given how rarely we see sig files containing multiple signatures, I would be inclined to not add the "new" sig file to the repo for now. At least its contents can always be derived from the signature file that *is* checked in.

Reply:
Sounds good, we should probably cleanup the temp file then at some point.

Reply:
The temp file does get cleaned up (indirectly), because it is created in `self._home`, which is tmp directory that is cleaned up later.